tests/PrefixCacheBootstrapperTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Support\Facades\Event;
+use Stancl\Tenancy\Bootstrappers\PrefixCacheTenancyBootstrapper;
+use Stancl\Tenancy\Events\TenancyEnded;
+use Stancl\Tenancy\Events\TenancyInitialized;
+use Stancl\Tenancy\Listeners\BootstrapTenancy;
+use Stancl\Tenancy\Listeners\RevertToCentralContext;
+
+beforeEach(function () {
+    config([
+        'tenancy.bootstrappers' => [PrefixCacheTenancyBootstrapper::class],
+        'cache.default' => 'redis',
+    ]);
+
+    Event::listen(TenancyInitialized::class, BootstrapTenancy::class);
+    Event::listen(TenancyEnded::class, RevertToCentralContext::class);
+});
+
+test('cache prefix is separate for each tenant', function () {
+    $originalPrefix = config('cache.prefix') . ':';
+
+    expect($originalPrefix)
+        ->toBe(app('cache')->getPrefix())
+        ->toBe(app('cache.store')->getPrefix());
+
+    $tenant1 = Tenant::create();
+    $tenant2 = Tenant::create();
+
+    $tenantOnePrefix = 'tenant_' . $tenant1->id . ':';
+
+    tenancy()->initialize($tenant1);
+    expect($tenantOnePrefix)
+        ->toBe(app('cache')->getPrefix())
+        ->toBe(app('cache.store')->getPrefix());
```

Comment:
This test should also try to write something to cache and then assert that the key is prefixed. Is there a way to fetch all keys from cache?

Reply:
> This test should also try to write something to cache and then assert that the key is prefixed.

Don't you think other tests are doing this where we are asserting cache data is separated? 

> Is there a way to fetch all keys from cache?
Not easy or Laravel provided but yes we can.

Reply:
> where we are asserting cache data is separated

Those are high-level tests — they assert that the data is separated, but we want to assert the low-level details as well (that the prefixes are correctly set).

Reply:
Looks like there is no easy way to get the keys. We need to get the store to get the keys. In our, I'm fetching from Redis `Redis::connection('cache')->keys('*')`. But these keys are not accurate. These also include the redis prefix.

Reply:
I see. Then I guess we can just try building the prefix manually and then try accessing the tenant's data from the central context and assert that it's the correct value

Reply:
Added https://github.com/archtechx/tenancy/pull/1014/commits/b920a2905ea6c0d3bab19c393e87fca0ab930323